src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,154 @@
+                    }
+                } while (presentInEntry != null);
+                table[indexInTable] = addEntry(key.hashCode(), key, value, table[indexInTable]);
+                size++;
+                resize();
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            if (table[0] != null) {
+                Entry<K, V> temp = table[0];
+                do {
+                    if (temp.key == null) {
+                        return temp.value;
+                    }
+                    temp = temp.next;
+                } while (temp != null);
+                return null;
+            } else {
+                return null;
+            }
+        } else {
+            int indexInTable = indexForEntry(key.hashCode(), table.length);
+            Entry<K, V> temp = table[indexInTable];
+            if (temp == null) {
+                return null;
+            }
+            do {
+                if (temp.key != null) {
+                    if (temp.key.hashCode() == key.hashCode() && key.equals(temp.key)) {
+                        return temp.value;
+                    }
+                }
+                temp = temp.next;
+            } while (temp != null);
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void transfer(Entry<K, V>[] newEntry) {
+        for (Entry<K, V> entry : table) {
+            if (entry != null) {
+                entry.copyTo(newEntry);
+                while (entry.next != null) {
+                    entry.next.copyTo(newEntry);
+                    entry.next = entry.next.next;
+                }
+            }
+        }
+    }
+
+    private void resize() {
+        if (size > table.length * LOAD_FACTOR) {
+            Entry<K, V>[] newTable = new Entry[table.length * 2];
+            transfer(newTable);
+            table = newTable;
+        }
+    }
+
+    private void putForNullKey(V value) {
+        if (table[0] != null) {
+            Entry<K, V> temp = table[0];
+            do {
+                if (temp.key == null) {
+                    temp.value = value;
+                    return;
+                }
+                if (temp.next == null) {
+                    break;
+                }
+                temp = temp.next;
+            } while (temp != null);
+            table[0] = addEntry(0, null, value, table[0]);
+            size++;
+            resize();
+
+        } else {
+            table[0] = addEntry(0, null, value, null);
+            size++;
+            resize();
+        }
+    }
+
+    private Entry<K, V> addEntry(int hashKay, K key, V value, Entry<K, V> next) {
+        Entry<K, V> entry = new Entry<>();
+        entry.key = key;
+        entry.value = value;
+        entry.next = next;
+        return entry;
+    }
+
+    public int indexForEntry(int hashCode, int capacity) {
```

Comment:
Don't we have access to capacity anyway?

Reply:
Done